tsdb/tombstones/tombstones.go

```diff
@@ -98,33 +97,27 @@ func WriteFile(logger log.Logger, dir string, tr Reader) (int64, error) {
 	buf.Reset()
 	// Write the meta.
 	buf.PutBE32(MagicTombstone)
-	buf.PutByte(tombstoneFormatV1)
 	n, err := f.Write(buf.Get())
 	if err != nil {
 		return 0, err
 	}
 	size += n
 
-	mw := io.MultiWriter(f, hash)
-
-	if err := tr.Iter(func(ref uint64, ivs Intervals) error {
-		for _, iv := range ivs {
-			buf.Reset()
+	bytes, err := Encode(tr)
+	if err != nil {
+		return 0, fmt.Errorf("error writing tombstones: %v", err)
+	}
 
-			buf.PutUvarint64(ref)
-			buf.PutVarint64(iv.Mint)
-			buf.PutVarint64(iv.Maxt)
+	// Ignore first byte which is the format type.
+	if _, err := hash.Write(bytes[1:]); err != nil {
+		return 0, fmt.Errorf("error writing tombstones: %v", err)
```

Comment:
```suggestion
		return 0, errors.Wrap(err, "calculating hash for tombstones")
```

Reply:
Wrap with name of what we are doing now, not in what function we are in - that's repetitive and not needed